### DISCUSSION TITLE
Auth within an organization?

### DISCUSSION THREAD

**Discussion by aleks-ivanov** [community/community#26759] [Actions]
<p>What is the best practice for authenticating operations between repos in an organization ? (e.g. one repo does an environment protected automation that populates data in another repo)</p>
<p>I know PAT from a high-level member of the organization does the trick, but this looks like a workaround, not a solution.</p>
<p>Is there a way to create some sort of an organization-wide access token (similar to PATs), to be used as a secret for a GiHA workflow, but without attaching it to any member rather attaching it to the organization itself ?</p>

**Comment by laughedelic** [comment, marked as answer]
<p>One way is to create a GitHub app for this. It can be</p>
<ul>
<li>hidden (visible only within the org),</li>
<li>have access to all repos in the org or only selected ones,</li>
<li>have restricted access permissions</li>
</ul>
<p>You would need to store a private key for this app as an org secret and use it in workflows to generate a short-lived token. Any operations made with this token will be on behalf of that app/bot because it has its own identity on GitHub.</p>
<p>It’s more involved to set up than using a PAT, but it definitely provides much more control and security.</p>
